### PR TITLE
support logging in CommandRunner when tests are executed in parallel

### PR DESCRIPTION
**Pull request by kartheekp-ms**
## Bug

Fixes: NuGet/Client.Engineering#613
Regression: No

## Fix

Details: When we run `NuGet.CommandLine.Tests`, every test in the test suite execute creates a `nuget.exe` process as shown below. Every `nuget.exe` process executing test command publishes `output and error stream` to the parent process so that test can check the result to verify the test status.

image

The way we read output and error stream from a Process while running tests in [`CommandRunner.Run`](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/CommandRunner.cs#L18) method doesn't work well when tests are executed in parallel because of following reasons.

- When multiple threads were trying to update the `output` or `error` string builder [here](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/CommandRunner.cs#L111) the results were inconsistent.
- [Task.WaitAll(outputTask, errorTask);](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/CommandRunner.cs#L95) sometime resulted in deadlock.

Work is in progress to support running test suites in parallel. see this [issue](https://github.com/NuGet/Client.Engineering/issues/67) for more info. This change is required to execute tests in parallel.

I found this [doc](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.beginoutputreadline?view=netcore-3.1) and [issue](https://github.com/dotnet/runtime/issues/27128#issuecomment-583010081) extremely helpful while working on this fix.

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  Fixing test utility method.
Validation:

### CODE REVIEW

File: test/TestUtilities/Test.Utility/CommandRunner.cs
Comment (kartheekp-ms) [inline test/TestUtilities/Test.Utility/CommandRunner.cs:25]: I found no callers in the entire codebase who is passing this optional parameter. Hence removed it. If there are any concerns from others, I can add it back.

File: test/TestUtilities/Test.Utility/CommandRunner.cs
Comment (nkolev92) [inline test/TestUtilities/Test.Utility/CommandRunner.cs:62]: Can you shortly elaborate in more detail how your change helps? 

I know you linked the issues and I appreciate that, but I was hoping you could save us some time digging through the linked docs if possible.

Comment (kartheekp-ms) [reply test/TestUtilities/Test.Utility/CommandRunner.cs:62]: Updated PR description with additional information. Pls let me know your feedback.

**Review by nkolev92 — APPROVED** [review]
Thanks for the link. 
Great find!

Consider adding a note on `what` not just the `why` in the PR title.

**Comment by kartheekp-ms** [comment]
🔔 @NuGet/nuget-client. Any feedback on this PR is helpful. Waiting for second approval.

**Review by zivkan — APPROVED** [review]
Great find!

**Pull request merged**
